### ISSUE TITLE
[receiver/windowsperfcountersreceiver] Windows Performance Counter Metrics should be of type float64, not int64

### ISSUE THREAD

**Issue by BinaryFissionGames** [open-telemetry/opentelemetry-collector-contrib#9136]
**Describe the bug**
In https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/ba8c7977b5b9f17900e326a07d76ecefbb205807 , the windowsperfcounters receiver was changed to emit metrics with an integral data type instead of a double data type. This change will lose specificity of some counter data.

The windowsperfcounters uses the PDH api (https://docs.microsoft.com/en-us/windows/win32/perfctrs/using-the-pdh-functions-to-consume-counter-data) to calculate displayable values. This API uses multiple datapoints for some counters (e.g. rate counters) in order to create a displayable value. These calculations include division operations, and therefore it isn't infrequent that they provide fractional values.

**Steps to reproduce**
Use the following config:
```yaml
receivers:
  windowsperfcounters:
    collection_interval: 10s
    metrics:
      processor.time:
        description: active and idle time of the processor
        unit: "%"
        gauge:
    perfcounters:
    - object: "Processor Information"
      instances: "*"
      counters:
        - name: "% Processor Time"
          metric: processor.time

exporters:
  logging:
    loglevel: debug

service:
  pipelines:
    metrics:
      receivers: [windowsperfcounters]
      exporters: [logging]
```

All exported values are integral

**What did you expect to see?**
Fractional values, like I see in perfmon

**What did you see instead?**
Integral values (e.g. 4, 2, 1, instead of values like 4.234)

**What version did you use?**

This is using commit 439369ef4

**Environment**
OS: Windows Server 2019
Compiler(if manually compiled): (e.g., "go 1.17.2")

Note: The analagous config below, with commit 9a5bf6e77, gives the expected fractional values:
```yaml
receivers:
  windowsperfcounters:
    collection_interval: 10s
    perfcounters:
    - object: "Processor Information"
      instances: "*"
      counters:
        "% Processor Time"

exporters:
  logging:
    loglevel: debug

service:
  pipelines:
    metrics:
      receivers: [windowsperfcounters]
      exporters: [logging]
```

**Comment by djaglowski** [comment]
Thanks for writing this up @BinaryFissionGames.

It looks like we need two things to address this, which I think can be separate issues:
1) Immediate fix - Don't cast values as ints.
2) Usability improvement - Add a `value_type` configuration option to the `metric` structs to allow individual metrics to be marked as ints.

**Comment by BinaryFissionGames** [comment]
@djaglowski I'll take care of the quick fix, changing the metric datapoints to be double.

**Comment by djaglowski** [comment]
Thanks @BinaryFissionGames.

@Mrod1598 I'm going to assign this to you, for the second part.

**Comment by djaglowski** [comment]
@BinaryFissionGames, I think this can be closed, but can you double check me?

**Comment by BinaryFissionGames** [comment]
@djaglowski yep, this was solved by #9138, closing as completed

**Comment by BinaryFissionGames** [comment]
> Thanks for writing this up @BinaryFissionGames.
> 
> It looks like we need two things to address this, which I think can be separate PRs:
> 
> 1. Immediate fix - Don't cast values as ints.
> 2. Usability improvement - Add a `value_type` configuration option to the `metric` structs to allow individual metrics to be marked as ints.

Looking back, the first one was solved, but the second one was not. If we want this functionality (`value_type` option), I think it should be a separate issue as a feature request.
